### PR TITLE
Add get_pipeline_cache_data

### PR DESCRIPTION
**Pull request by zakorgy**
This change is [Reviewable](https://reviewable.io/reviews/maikklein/ash/166)

### CODE REVIEW

**Comment by zakorgy** [comment]
@MaikKlein I have updated the PR according to your advice. I can squash the commits if you want.

File: ash/src/device.rs [hunk 1]
```diff
@@ -1276,6 +1276,36 @@ pub trait DeviceV1_0 {
         }
     }
 
+    unsafe fn get_pipeline_cache_data(
+        &self,
+        pipeline_cache: vk::PipelineCache,
+    ) -> VkResult<Vec<u8>> {
+        let data_size: *mut usize = mem::uninitialized();
+        let data: *mut c_void = mem::uninitialized();
```
Comment (MaikKlein) [inline ash/src/device.rs]: I think you should initialize them explicitly to a null ptr.

File: ash/src/device.rs [hunk 2]
```diff
@@ -1276,6 +1276,36 @@ pub trait DeviceV1_0 {
...
+
+        let err_code = self.fp_v1_0().get_pipeline_cache_data(
+            self.handle(),
+            pipeline_cache,
+            data_size,
+            data,
+        );
+
+        match err_code {
+            vk::Result::SUCCESS => Ok(Vec::from_raw_parts(data as _, *data_size, *data_size)),
```
Comment (MaikKlein) [inline ash/src/device.rs]: You have to allocate the buffer to `data` yourself. You can do this with `Vec::with_capacity` and `set_len`. And you need to  call `get_pipeline_cache_data` two times to get the size, and then to fill the buffer.

**Comment by kvark** [comment]
@MaikKlein any chance of releasing a version with this?

**Comment by MaikKlein** [comment]
@kvark Oh I can't believe I haven't pushed a new version. I fix up a few PRs and I publish a new version later today.

**Pull request merged**